### PR TITLE
CLN: assorted tslibs cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -1325,7 +1325,7 @@ cdef class BusinessDay(BusinessMixin):
                 off_str += str(td.microseconds) + "us"
             return off_str
 
-        if isinstance(self.offset, timedelta):
+        if PyDelta_Check(self.offset):
```
Comment (jorisvandenbossche) [inline pandas/_libs/tslibs/offsets.pyx:1328]: Cython doesn't optimize this automatically? 
(and does this make a noticeable difference?)

Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx:1328]: cython does not do this automatically.  it makes a small differnece, but enough that we do this in the rest of the cython code

Comment (jorisvandenbossche) [reply pandas/_libs/tslibs/offsets.pyx:1328]: OK

Comment (jreback) [reply pandas/_libs/tslibs/offsets.pyx:1328]: yeah we use this type of check with datetime types a lot

**Review by jreback — APPROVED** [review]

**Pull request merged**